flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolBatchSlotRequestTest.java

```diff
@@ -147,14 +148,42 @@ public void testPendingBatchSlotRequestDoesNotFailIfAllocationFails() throws Exc
 
 			final CompletableFuture<PhysicalSlot> slotFuture = SlotPoolUtils.requestNewAllocatedBatchSlot(slotPool, directMainThreadExecutor, resourceProfile);
 
-			SlotPoolUtils.failAllocation(slotPool, directMainThreadExecutor, allocationIdFuture.get());
+			SlotPoolUtils.failAllocation(slotPool, directMainThreadExecutor, allocationIdFuture.get(), new FlinkException("Failed request"));
 
 			assertThat(slotFuture.isDone(), is(false));
 		}
 	}
 
 	/**
-	 * Tests that a batch slot request won't fail if its resource manager request fails.
+	 * Tests that a batch slot request does react to {@link SlotPool#failAllocation(AllocationID, Exception)}
+	 * signals whose exception is {@link UnfulfillableSlotRequestException}.
+	 */
+	@Test
+	public void testPendingBatchSlotRequestFailsIfAllocationFailsUnfulfillable() throws Exception {
+		final TestingResourceManagerGateway testingResourceManagerGateway = new TestingResourceManagerGateway();
+		final CompletableFuture<AllocationID> allocationIdFuture = new CompletableFuture<>();
+		testingResourceManagerGateway.setRequestSlotConsumer(slotRequest -> allocationIdFuture.complete(slotRequest.getAllocationId()));
+
+		final ComponentMainThreadExecutor directMainThreadExecutor = ComponentMainThreadExecutorServiceAdapter.forMainThread();
+
+		final Time batchSlotTimeout = Time.milliseconds(1000L);
```

Comment:
No need to set this timeout. Per default this should be 2s if using the `SlotPoolBuilder`.